src/test/java/com/rocketden/main/api/ProblemTests.java

```diff
@@ -1,136 +1,138 @@
 package com.rocketden.main.api;
 
-import static org.assertj.core.api.Assertions.assertThat;
-import static org.hamcrest.Matchers.containsString;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
 import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
-import javax.sql.DataSource;
-
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.rocketden.main.dto.problem.CreateProblemRequest;
 import com.rocketden.main.dto.problem.ProblemDto;
+import com.rocketden.main.exception.ProblemError;
+import com.rocketden.main.exception.api.ApiError;
+import com.rocketden.main.exception.api.ApiErrorResponse;
 import com.rocketden.main.util.UtilityTestMethods;
 
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.test.web.client.TestRestTemplate;
-import org.springframework.boot.web.server.LocalServerPort;
 import org.springframework.http.MediaType;
 import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
 import org.springframework.transaction.annotation.Transactional;
 
-import org.junit.Assert;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
 
-@SpringBootTest(
-    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
-    properties = "spring.datasource.type=com.zaxxer.hikari.HikariDataSource"
-)
+@SpringBootTest(properties = "spring.datasource.type=com.zaxxer.hikari.HikariDataSource")
```

Comment:
What was your thinking behind this update?

Reply:
Since we're now just using MockMvc instead of the RestTemplate/port/localhost stuff, I think that allows us to get rid of the webEnvironment annotation - the tests work so I think this is correct, but I haven't looked into it deeply so lmk if I'm wrong about that haha

Reply:
No I think you're right too, thanks for the explanation 🙂